keyboards/install.php

```diff
@@ -90,35 +90,48 @@ protected static function WriteWindowsBoxes() {
       $bcp47 = rawurlencode(empty(self::$bcp47) ? '' : self::BOOTSTRAP_SEPARATOR.self::$bcp47);
       $tier = self::$tier;
       $version = self::$versions->windows->$tier;
-      $e_keyboard_id = rawurlencode($keyboard->id);
-      $h_keyboard_name = htmlentities($keyboard->name);
 
-      // Note, we don't need to capture an event for the keyboard download here, because we'll capture it when the bootstrap downloads the file.
-      $downloadLink = KeymanHosts::Instance()->downloads_keyman_com . "/windows/{$tier}/{$version}/keyman-setup" . self::BOOTSTRAP_SEPARATOR . "{$keyboard->id}{$bcp47}.exe";
-      $downloadLinkE = json_encode($downloadLink, JSON_UNESCAPED_SLASHES);
+      $e = [
+        'name' => $keyboard->name,
+        'host' => KeymanHosts::Instance()->downloads_keyman_com,
+        'tier' => $tier,
+        'version' => $version,
+        'id' => $keyboard->id,
+        'bcp47' => empty(self::$bcp47) ? '' : self::$bcp47
+      ];
+
+      $u = array_map('rawurlencode', $e);
+      $hu = array_map('htmlentities', $u);
+      $h = array_map('htmlentities', $e);
+
+      // Note, we don't need to capture an event for the keyboard download here, because we'll capture
+      // it when the bootstrap downloads the file.
+
+      // This maps to buildStandardWindowsDownloadUrl in install.js (which we don't use here for those
+      // few users who have JS disabled -- although this is not really a tested/supported scenario)
```

Comment:
Can this happen at all? If users don't have JS enabled this doesn't show because `keyboard-details.js` won't set the `data-platform` attribute.

Reply:
True... it doesn't hurt, though, and we can use a similar pattern later when we do the 'unknown platform' landing page (which is what appears if JS is disabled).